### PR TITLE
Filter device classes per platform when registering missing devices

### PR DESCRIPTION
**Pull request by Bence1001**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context

The step ensures all the Bitrise test devices are registered on the Apple Developer Portal and ensures the test devices are included in the Step-managed Provisioning Profile.

Bitrise test devices can contain any type of device (macOS, tvOS, ....), however not all types of devices can be used in each Profile type (e.g.: iOS Profile can not be used to provision a tvOS app). 

The Step did not filter the compatible devices, which can result in false-negative Profile checks (e.g.: a tvOS Xcode managed Profile was marked as invalid because it did not include an iPhone test device registered on Bitrise).

Resolves: https://bitrise.atlassian.net/browse/STEP-6

### Changes

- **Filter test devices based on the platform when missing devices are registered.**

|        | iOS | tvOS | MacOS* |
|--------|-----|------|------|
| iPhone | ✅   | ❌    | ❌    |
| iPod   | ✅   |  ❌    | ❌    |
| iPad   | ✅   | ❌    | ❌    |
| Watch  | ✅   | ❌    | ❌    |
| Apple TV   | ❌   | ✅    | ❌    |

*MacOS projects are not supported by the step

- Device-related codes moved to devices.go, while related tests to devices_test.go file.
- The code block solving test device registration on the Apple Developer Portal, listing and filtering the Devices moved to the `listRelevantDevPortalDevices` function.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -237,165 +236,36 @@ func TestDownloadLocalCertificates(t *testing.T) {
...
-	got, err := registerMissingDevices(args.client, args.bitriseDevices, args.devportalDevices)
-	require.NoError(t, err, "registerMissingDevices()")
-	require.Equal(t, want, got, "registerMissingDevices()")
-	mockClient.AssertExpectations(t)
-}
-
 func Test_createWildcardBundleID(t *testing.T) {
 	tests := []struct {
-		name    string
+		name     string
```
Comment (ofalvai) [inline main_test.go:241]: These whitespace changes look weird, is this intentional?

File: main.go [hunk 2]
```diff
@@ -615,11 +537,16 @@ func main() {
...
-			if err != nil {
-				failf("Failed to add devices registered on Bitrise to Developer Portal: %s", err)
-			}
-			devices = append(devices, registeredDevices...)
+			testDevices = conn.TestDevices
+		}
+
+		devPortalDevices, err := listRelevantDevPortalDevices(client, testDevices, platform)
+		if err != nil {
+			failf(err.Error())
```
Comment (ofalvai) [inline main.go:545]: Why is this an empty error?

File: main_test.go (hunk 1 above)
Comment (Bence1001) [reply main_test.go:241]: They are there to align the rest of the types with `bundleID`'s `string`. Probably the result of a previously forgotten `gofmt`.

File: devices.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		log.Debugf("- %s, %s, UDID (%s), ID (%s)", devPortalDevice.Attributes.Name, devPortalDevice.Attributes.DeviceClass, devPortalDevice.Attributes.UDID, devPortalDevice.ID)
+	}
+
+	if len(testDevices) != 0 {
+		log.Infof("Checking if %d Bitrise test device(s) are registered on Apple Developer Portal", len(testDevices))
+		for _, testDevice := range testDevices {
+			log.Debugf("- %s, %s, UDID (%s), added at %s", testDevice.Title, testDevice.DeviceType, testDevice.DeviceID, testDevice.UpdatedAt)
+		}
+
+		devPortalDevice, err := registerMissingTestDevices(client, testDevices, devPortalDevices)
```
Comment (ofalvai) [inline devices.go:134]: This seems like a major side-effect for a method called `listRelevantDevPortalDevices()` 😵‍💫

**Review by ofalvai — APPROVED** [review]

**Review by ofalvai — APPROVED** [review]

**Pull request merged**